apps/ledger-live-mobile/src/components/WebPlatformPlayer/WebViewV2.tsx

```diff
@@ -62,21 +36,234 @@ import { track } from "../../analytics/segment";
 import prepareSignTransaction from "./liveSDKLogic";
 import { StackNavigatorNavigation } from "../RootNavigator/types/helpers";
 import { BaseNavigatorStackParamList } from "../RootNavigator/types/BaseNavigator";
+import { analyticsEnabledSelector } from "../../reducers/settings";
+import packageInfo from "../../../package.json";
 
 const tracking = trackingWrapper(track);
 
-type Props = {
-  manifest: AppManifest;
-  inputs?: Record<string, string>;
-};
+function useWebViewHandlers({
+  onMessageRaw,
+  onReloadRaw,
+  setLoadDate,
+}: {
+  onMessageRaw: ReturnType<typeof useWalletAPIServer>["onMessage"];
+  onReloadRaw: ReturnType<typeof useWalletAPIServer>["onReload"];
+  setLoadDate: (date: Date) => void;
+}) {
+  const onMessage = useCallback(
+    e => {
+      if (e.nativeEvent?.data) {
+        onMessageRaw?.(e.nativeEvent.data);
+      }
+    },
+    [onMessageRaw],
+  );
+
+  const onReload = useCallback(() => {
+    onReloadRaw();
+    setLoadDate(new Date()); // TODO: wtf
+  }, [onReloadRaw, setLoadDate]);
+
+  return { onMessage, onReload };
+}
+
+function useUiHook(): Partial<UiHook> {
+  const navigation = useNavigation();
+
+  return useMemo(
+    () => ({
+      "account.request": ({ accounts$, currencies, onSuccess, onError }) => {
+        if (currencies.length === 1) {
+          navigation.navigate(NavigatorName.RequestAccount, {
+            screen: ScreenName.RequestAccountsSelectAccount,
+            params: {
+              accounts$,
+              currency: currencies[0],
+              allowAddAccount: true,
+              onSuccess,
+              onError,
+            },
+          });
+        } else {
+          navigation.navigate(NavigatorName.RequestAccount, {
+            screen: ScreenName.RequestAccountsSelectCrypto,
+            params: {
+              accounts$,
+              currencies,
+              allowAddAccount: true,
+              onSuccess,
+              onError,
+            },
+          });
+        }
+      },
+      "account.receive": ({
+        account,
+        parentAccount,
+        accountAddress,
+        onSuccess,
+        onCancel,
+        onError,
+      }) => {
+        navigation.navigate(ScreenName.VerifyAccount, {
+          account,
+          parentId: parentAccount ? parentAccount.id : undefined,
+          onSuccess: () => onSuccess(accountAddress),
+          onClose: onCancel,
+          onError,
+        });
+      },
+      "message.sign": ({ account, message, onSuccess, onError, onCancel }) => {
+        navigation.navigate(NavigatorName.SignMessage, {
+          screen: ScreenName.SignSummary,
+          params: {
+            message,
+            accountId: account.id,
+            onConfirmationHandler: onSuccess,
+            onFailHandler: onError,
+          },
+          onClose: onCancel,
+        });
+      },
+      "transaction.sign": ({
```

Comment:
Why does mobile only have `"transaction.sign"` and not `"transaction.signAndBrodcast"` as well anymore?

Reply:
`transaction.broadcast` is a UI that triggers when broadcast is done (optimistic UI feedback). AFAIK, currently it only exists in desktop. Broadcast logic is already handled within the custom hook in live-common. The purpose of this refactoring is to extract all common logic from UI and leave only platform specific UI implementation in desktop/mobile workspace.

Reply:
transaction.broadcast is a UI that triggers when broadcast is done (optimistic UI feedback). AFAIK, currently it only exists in desktop. Broadcast logic is already handled within the custom hook in live-common. The purpose of this refactoring is to extract all common logic from UI and leave only platform specific UI implementation in desktop/mobile workspace.